src/prefect/tasks/prefect/flow_run.py

```diff
@@ -43,6 +47,7 @@ def __init__(
         wait: bool = False,
         new_flow_context: dict = None,
         run_name: str = None,
+        idempotency_key: str = None,
```

Comment:
I don't think I agree with this change - if utilized, it means that one and only one flow run will ever be created across all runs of the parent flow, which can always be achieved with a hardcoded string, but this feels like it will invite confusion if people use it

Reply:
I don't follow? Parameters can be passed to tasks two ways:

- As static parameters to the constructor
- As dynamic parameters to `__call__`/`bind` (usually from tasks)

I don't see a good reason to restrict a parameter to one or the other, and frequently find users confused when a parameter is only available in the constructor (or as part of `.run`). Restricting to a `run` parameter doesn't stop a user from incorrectly using `idempotency_key`, and the default behavior remains `None` as before.

Reply:
Oh wait, I see what you're saying. Sure, I can remove it.

Reply:
I think there's a good reason to restrict whenever a static value for something is misleading or not particularly useful; if I built a flow and used a static key here, then what's the point of the task?  It will create a single flow run once, and never create a second one.  Users will almost certainly ask "Why is my task not creating new runs"?  For users that (for some reason) actually _do_ want that functionality, they can opt-in via a hardcoded static string in the run method, but at least in that case they'll know what they're getting themselves into.